kornia/losses/focal.py

```diff
@@ -198,18 +204,25 @@ def binary_focal_loss_with_logits(
             stacklevel=2,
         )
 
-    if not isinstance(input, torch.Tensor):
-        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
+    if not isinstance(input, Tensor):
+        raise TypeError(f"Input type is not a Tensor. Got {type(input)}")
 
     if not len(input.shape) >= 2:
         raise ValueError(f"Invalid input shape, we expect BxCx*. Got: {input.shape}")
 
     if input.size(0) != target.size(0):
         raise ValueError(f'Expected input batch_size ({input.size(0)}) to match target batch_size ({target.size(0)}).')
 
+    if pos_weight is None:
+        pos_weight = torch.ones(input.size(-1), device=input.device, dtype=input.dtype)
+
+    KORNIA_CHECK_IS_TENSOR(pos_weight)
+    KORNIA_CHECK(input.size(-1) == pos_weight.size(0), "Expected pos_weight equals number of "
+                                                       "classes.")
+
     probs_pos = torch.sigmoid(input)
     probs_neg = torch.sigmoid(-input)
-    loss_tmp = -alpha * torch.pow(probs_neg, gamma) * target * F.logsigmoid(input) - (
+    loss_tmp = -alpha * pos_weight * torch.pow(probs_neg, gamma) * target * F.logsigmoid(input) - (
```

Comment:
I replaced `F.log_softmax(input, dim=1)` with `input.log_softmax(1)`

Unfortunately Tensor doesn't exist method logsigmoid(), so to avoid using nn.functional, I replaced it with `input.sigmoid().log()` which the same and `-input.sigmoid().log()`

Reply:
So finally I deleted `import torch.nn.functional as F`